internal/test/e2e/registryMirror.go

```diff
@@ -39,6 +39,20 @@ func (e *E2ESession) setupRegistryMirrorEnv(testRegex string) error {
 		caCert = e.testEnvVars[e2etests.RegistryCACertTinkerbellVar]
 	}
 
+	re = regexp.MustCompile(`^.*VSphere.*Authenticated.*$`)
+	if re.MatchString(testRegex) {
+		endpoint = e.testEnvVars[e2etests.PrivateRegistryEndpointVar]
+		port = e.testEnvVars[e2etests.PrivateRegistryPortVar]
+		caCert = e.testEnvVars[e2etests.PrivateRegistryCACertVar]
+	}
+
+	re = regexp.MustCompile(`^.*Tinkerbell.*Authenticated.*$`)
+	if re.MatchString(testRegex) {
```

Comment:
these conditions should be mutually exclusive. Can you use `else if`s here to avoid unnecessary regex matches?